tests/Issues/Issue9388/Rule/ExtbaseAnnotationToAttributeRector.php

```diff
@@ -0,0 +1,185 @@
+
+class MyEntity
+{
+    /**
+     * @Extbase\ORM\Transient()
+     */
+    protected string $myProperty;
+}
+CODE_SAMPLE
+            ,
+            <<<'CODE_SAMPLE'
+use TYPO3\CMS\Extbase\Annotation as Extbase;
+
+class MyEntity
+{
+    #[Extbase\ORM\Transient()]
+    protected string $myProperty;
+}
+CODE_SAMPLE
+        )]);
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $phpDocInfo = $this->phpDocInfoFactory->createFromNode($node);
+        if (! $phpDocInfo instanceof PhpDocInfo) {
+            return null;
+        }
+
+        $uses = $this->useImportsResolver->resolveBareUses();
+        $annotationAttributeGroups = $this->processDoctrineAnnotationClasses($phpDocInfo, $uses);
+        if ($annotationAttributeGroups === []) {
+            return null;
+        }
+
+        $this->docBlockUpdater->updateRefactoredNodeWithPhpDocInfo($node);
+
+        foreach ($annotationAttributeGroups as $attributeGroup) {
+            foreach ($attributeGroup->attrs as $attr) {
+                $phpAttributeName = $attr->name->getAttribute(AttributeKey::PHP_ATTRIBUTE_NAME);
+                $this->attributeDecorator->decorate($phpAttributeName, $attr);
+            }
+        }
+
+        $node->attrGroups = \array_merge($node->attrGroups, $annotationAttributeGroups);
+        return $node;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    /**
+     * @param Use_[] $uses
+     * @return AttributeGroup[]
+     */
+    private function processDoctrineAnnotationClasses(PhpDocInfo $phpDocInfo, array $uses): array
+    {
+        if ($phpDocInfo->getPhpDocNode()->children === []) {
+            return [];
+        }
+
+        $doctrineTagAndAnnotationToAttributes = [];
+        $doctrineTagValueNodes = [];
+        foreach ($phpDocInfo->getPhpDocNode()->children as $phpDocChildNode) {
+            if (! $phpDocChildNode instanceof PhpDocTagNode) {
+                continue;
+            }
+
+            if (! $phpDocChildNode->value instanceof DoctrineAnnotationTagValueNode) {
+                continue;
+            }
+
+            $doctrineTagValueNode = $phpDocChildNode->value;
+            $annotationToAttribute = $this->matchAnnotationToAttribute($doctrineTagValueNode);
+            if (! $annotationToAttribute instanceof AnnotationToAttribute) {
+                continue;
+            }
+
+            // Fix the missing leading slash in most of the wild use cases
+            if (str_starts_with($doctrineTagValueNode->identifierTypeNode->name, '@TYPO3\CMS')) {
+                $doctrineTagValueNode->identifierTypeNode->name = str_replace(
+                    '@TYPO3\CMS',
+                    '@\\TYPO3\CMS',
+                    $doctrineTagValueNode->identifierTypeNode->name
+                );
+            }
```

Comment:
This is why when you flip the rule, the rule is "working", while actually invalid, as the original namespaced name is current namespace + value identifier if no `\\` prefix, so you're changing behaviour of valid identifier. 

Add `use TYPO`, and you will have correct detection :)